code/AssetLib/Blender/BlenderLoader.cpp

```diff
@@ -1338,4 +1295,55 @@ aiNode *BlenderImporter::ConvertNode(const Scene &in, const Object *obj, Convers
     return node.release();
 }
 
+std::string BlenderImporter::ParseMagicToken(std::shared_ptr<IOStream>& stream) const {
```

Comment:
You should not pass `shared_ptr` via reference. Either pass by value or pass the raw pointer.

Reply:
In this specific case the stream is an input/output parameter, as it can internally be reset, if the file is compressed, and will be reused outside of this function. So passing by value wouldn't work. Passing by raw pointer would require us to clean up the old memory, so the whole point of using smart pointers is lost.

Of course we could rewrite more of the code here, but I didn't want to touch too much of the original logic (besides converting exceptions to strings).